### PR TITLE
Bundle types in `index.d.ts`

### LINKED ISSUE #244: `meow(...).flags` is `any` in 12.1.0
With version <https://github.com/sindresorhus/meow/tree/v12.1.0> `type-fest` is not declared as a dependency anymore in the actual published package. `type-fest` then can't be resolved in [index.d.ts](https://github.com/sindresorhus/meow/blob/b412c0d78702f3fb2fff4a474007422bc78a63f9/index.d.ts#L1-L4) and `meow(...).flags` is just `any`.

I have created a repository to demonstrate the issue
- <https://github.com/melusc/meow-dep-less-issue/>
- <https://github.com/melusc/meow-dep-less-issue/actions/runs/5989672259/job/16246105667>

### PR DESCRIPTION
**Pull request by tommy-mitchell**
Fixes #244.

Uses [`rollup-plugin-dts`](https://github.com/Swatinem/rollup-plugin-dts) to bundle types and adds another `tsd` test for the bundled types.

### CODE REVIEW

**Comment by tommy-mitchell** [comment]
I pinned the version of `type-fest`. In v4.3.0, the build fails due to https://github.com/sindresorhus/type-fest/issues/674.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/type-fest/releases/tag/v4.3.1

File: package.json [hunk 1]
```diff
@@ -22,11 +22,10 @@
 	"scripts": {
 		"prepare": "npm run build",
 		"build": "rollup --config",
-		"test": "xo && tsd && npm run build && ava"
+		"test": "xo && npm run build && ava && tsd -t build/index.d.ts"
```
Comment (sindresorhus) [inline package.json:25]:
```suggestion
		"test": "xo && npm run build && ava && tsd --typings build/index.d.ts"
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by tommy-mitchell** [comment]
Resolved.

**Comment by jaywcjlove** [comment]
image

Could not read type file anymore. @tommy-mitchell @sindresorhus 

```js
"typescript": "^5.2.2"
```

**Comment by tommy-mitchell** [comment]
@sindresorhus do we need a `types` field in `package.json`? Not really sure why NPM isn't resolving it.

**Comment by jaywcjlove** [comment]
> @sindresorhus do we need a `types` field in `package.json`? Not really sure why NPM isn't resolving it.

used in a project, the type could not be loaded @tommy-mitchell

**Comment by jaywcjlove** [comment]
@sindresorhus thx!

**Pull request merged**